### PR TITLE
Java serialization can be configured to handled by the configured global serializer

### PR DESCRIPTION
**Pull request by asimarslan**
fixes https://github.com/hazelcast/hazelcast/issues/6790

### CODE REVIEW

**Comment by mdogan** [comment]
@asimarslan: You should handle Spring config too :)

**Comment by hasancelik** [comment]
run-lab-run

**Comment by sancar** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/config/GlobalSerializerConfig.java [hunk 1]
```diff
@@ -47,11 +49,20 @@ public GlobalSerializerConfig setImplementation(Serializer implementation) {
         return this;
     }
 
+    public boolean isOverrideJavaSerialization() {
+        return overrideJavaSerialization;
+    }
+
+    public void setOverrideJavaSerialization(boolean overrideJavaSerialization) {
+        this.overrideJavaSerialization = overrideJavaSerialization;
```
Comment (cangencer) [inline hazelcast/src/main/java/com/hazelcast/config/GlobalSerializerConfig.java]: should use fluent config here

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/config/GlobalSerializerConfig.java]: +1

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java [hunk 2]
```diff
@@ -287,13 +290,19 @@ public final void register(Class type, Serializer serializer) {
     }
 
     public final void registerGlobal(final Serializer serializer) {
+        registerGlobal(serializer, false);
+    }
+
+    public final void registerGlobal(final Serializer serializer, boolean overrideJavaSerialization) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java]: Imho we should pull out the serialization resolution into a separate data-structure to make the SerializationService a bit cleaner. But not part of this PR.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultSerializationServiceBuilder.java [hunk 3]
```diff
@@ -235,7 +235,8 @@ public SerializationService build() {
                     ((HazelcastInstanceAware) serializer).setHazelcastInstance(hazelcastInstance);
                 }
 
-                ((AbstractSerializationService) ss).registerGlobal(serializer);
+                ((AbstractSerializationService) ss)
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultSerializationServiceBuilder.java]: I would pull out a local variable.

AbstractSerializationService ss = ....
ss.registerGlobal(...)

Else the code is so hard to digest.
